r-lib-error-1.0.md

```diff
@@ -0,0 +1,64 @@
+
+# Specification of r-lib errors, version 1.0.0
+
+Note: this is an experimental draft.
+
+Note: version 1.0.0 corresponds to the `rlang_error` class, as
+implemented in rlang 0.4.0.
+
+## Classes
+
+An r-lib error object always has at least classes `rlib_error`, `error`
+and `condition`. It may have other classes.
+
+## Members
+
+An `rlib_error` object has the following members:
+
+* `message`: `character(1)`, optional, the error message. If this is
+  missing then the `conditionMessage()` S3 method must be able to generate
+  the error message for the class.
+* `call`: optional, the captured call, of mode `call`.
+  Even if this is missing, the `conditionCall()` S3 method might still
+  generate the call.
+* `trace`: optional, an `rlib_trace` object, see below. 
+* `parent`: optional, pointer to a parent error, of class `error`. It
+  may also be `NULL`, which is the same as missing.
+* `version`: optional, `rlib_error` version, defaults to 1.0.0, i.e. this
+  specification.
+
+## Traces
+
+An `rlib_trace` object has the following members:
```

Comment:
I think you should define the base type and class here too.

Reply:
OK.